awx/api/serializers.py

```diff
@@ -817,6 +821,18 @@ def to_representation(self, obj):
         ret['job_explanation'] = _(obj.job_explanation)
         return ret
 
+    def get_launched_by(self,obj):
+        launched_by = obj.launched_by
+        if launched_by is not None:
+            return model_to_dict(launched_by)
```

Comment:
This is pretty tall:

```
{'allow_simultaneous': False,
 'artifacts': '{}',
 'become_enabled': False,
 'cancel_flag': False,
 'canceled_on': None,
 'celery_task_id': '6a361f85-c67c-4799-a1ff-1fdf1952e488',
 'controller_node': '',
 'created': '2021-03-15 16:37:38.703310+00:00',
 'created_by': 'ryan',
 'custom_virtualenv': '/var/lib/awx/venv/ansible',
 'dependencies_processed': True,
 'description': '',
 'diff_mode': False,
 'elapsed': '0.000',
 'emitted_events': 9,
 'execution_environment': 'Default EE-1',
 'execution_node': 'awx_1',
 'extra_vars': '{}',
 'failed': False,
 'finished': '2021-03-15 16:38:10.021148+00:00',
 'force_handlers': False,
 'forks': 0,
 'id': 1,
 'installed_collections': '{"/usr/share/ansible/collections/ansible_collections": {"awx.awx": {"version": "17.1.0"}, "amazon.aws": {"version": "1.4.1"}, "ovirt.ovirt": {"version": "1.2.5"}, "google.cloud": {"version": "1.0.2"}, "ansible.posix": {"version": "1.2.0"}, "openstack.cloud": {"version": "1.0.1"}, "community.vmware": {"version": "1.8.0"}, "azure.azcollection": {"version": "1.4.0"}, "theforeman.foreman": {"version": "2.0.1"}, "community.kubernetes": {"version": "1.2.0"}}}',
 'instance_group': 'tower-1',
 'inventory': 'Demo Inventory-1',
 'job_args': '["podman", "run", "--rm", "--tty", "--interactive", "--workdir", "/runner/project", "-v", "/tmp/awx_1_4x36afff:/runner:Z", "--env-file", "/tmp/awx_1_4x36afff/artifacts/1/env.list", "--quiet", "--name", "ansible_runner_1", "--user=root", "quay.io/ansible/awx-ee", "ansible-playbook", "-u", "admin", "-i", "/runner/inventory/hosts", "-e", "@/runner/env/extravars", "hello_world.yml"]',
 'job_cwd': '/runner/project',
 'job_env': '{"LAUNCHED_BY_RUNNER": "1", "AWX_ISOLATED_DATA_DIR": "/runner/artifacts/1", "ANSIBLE_FORCE_COLOR": "True", "ANSIBLE_HOST_KEY_CHECKING": "False", "ANSIBLE_INVENTORY_UNPARSED_FAILED": "True", "ANSIBLE_PARAMIKO_RECORD_HOST_KEYS": "False", "ANSIBLE_VENV_PATH": "/var/lib/awx/venv/ansible", "AWX_PRIVATE_DATA_DIR": "/tmp/awx_1_4x36afff", "JOB_ID": "1", "INVENTORY_ID": "1", "PROJECT_REVISION": "347e44fea036c94d5f60e544de006453ee5c71ad", "ANSIBLE_RETRY_FILES_ENABLED": "False", "MAX_EVENT_RES": "700000", "AWX_HOST": "https://towerhost", "ANSIBLE_SSH_CONTROL_PATH_DIR": "/runner/cp", "ANSIBLE_COLLECTIONS_PATHS": "/runner/requirements_collections:~/.ansible/collections:/usr/share/ansible/collections", "ANSIBLE_ROLES_PATH": "/runner/requirements_roles:~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles", "ANSIBLE_STDOUT_CALLBACK": "awx_display", "RUNNER_OMIT_EVENTS": "False", "RUNNER_ONLY_FAILED_EVENTS": "False"}',
 'job_explanation': '',
 'job_slice_count': 1,
 'job_slice_number': 0,
 'job_tags': '',
 'job_template': 'Demo Job Template-7',
 'job_type': 'run',
 'launch_type': 'manual',
 'limit': '',
 'modified': '2021-03-15 16:38:04.536258+00:00',
 'modified_by': None,
 'name': 'Demo Job Template',
 'old_pk': None,
 'organization': 'Default-1',
 'playbook': 'hello_world.yml',
 'polymorphic_ctype': 'job',
 'project': 'Demo Project-6',
 'project_update': None,
 'result_traceback': '',
 'schedule': None,
 'scm_branch': '',
 'scm_revision': '347e44fea036c94d5f60e544de006453ee5c71ad',
 'skip_tags': '',
 'start_args': 'hidden',
 'start_at_task': '',
 'started': '2021-03-15 16:38:04.612354+00:00',
 'status': 'successful',
 'survey_passwords': '{}',
 'timeout': 0,
 'unified_job_template': 'Demo Job Template-7',
 'unifiedjob_ptr': '2021-03-15 16:37:38.703310+00:00-1-successful',
 'use_fact_cache': False,
 'verbosity': 0,
 'webhook_credential': None,
 'webhook_guid': '',
 'webhook_service': ''}
```

Is this the data we want to surface here?

Reply:
Probably not.  We need the type, the name, and the id, IIRC.

Reply:
@ryanpetrello yeah, the api url sounds like a reasonable thing to want, for consistency.  The immediate use of these additions, though, is to construct a UI route.

Reply:
If we did something _like this_, though, we could represent things like system dependencies like:

```
{
    "type": "dependency",
    "id": null,
    "name": "System Dependency",
    "url": null
}
```

Reply:
:eyes: Yeah if we could trim this down and have consistent contents that would be ideal

so instead of `model_to_dict` we would have

dependency
```
    "type": "dependency",
    "id": null,
    "name": "System Dependency",
    "url": null
```
workflow triggered job
```
    "type": "workflow",
    "id": 123,
    "name": "Workflow that triggered this job Name",
    "url": /api/v2/workflows/123
```

webhook
```
    "type": "job_template",
    "id": 345,
    "name": "Job Template That Has the Webook Name",
    "url": /api/v2/job_template/345
```

... etc

Reply:
I have made these changes. 
`launched_by` returns 4 attributes : 
id , type , name , url

Right now the url point towards the API endpoints as mentioned above.

Reply:
cc @ryanpetrello @kdelee @jbradberry